benchflow-experiment-manager/application/src/test/java/cloud/benchflow/experimentmanager/BenchFlowExperimentManagerApplicationTestModeIT.java

```diff
@@ -0,0 +1,231 @@
+import java.util.concurrent.TimeUnit;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.core.Response;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+/**
+ * @author Jesper Findahl (jesper.findahl@gmail.com) created on 2017-06-04
+ */
+public class BenchFlowExperimentManagerApplicationTestModeIT
+    extends DockerComposeIT {
+
+  private static final int TEST_PORT = 8080;
+  private static final String TEST_ADDRESS = "localhost:" + TEST_PORT;
+
+  @Rule
+  public final DropwizardAppRule<BenchFlowExperimentManagerConfiguration> RULE =
+      new DropwizardAppRule<>(BenchFlowExperimentManagerApplication.class, "../configuration.yml",
+          ConfigOverride.config("mongoDB.hostname", MONGO_CONTAINER.getIp()),
+          ConfigOverride.config("mongoDB.port", String.valueOf(MONGO_CONTAINER.getExternalPort())),
+          ConfigOverride.config("minio.address",
+              "http://" + MINIO_CONTAINER.getIp() + ":" + MINIO_CONTAINER.getExternalPort()),
+          ConfigOverride.config("minio.accessKey", MINIO_ACCESS_KEY),
+          ConfigOverride.config("minio.secretKey", MINIO_SECRET_KEY),
+          ConfigOverride.config("driversMaker.address", TEST_ADDRESS),
+          ConfigOverride.config("testManager.address", TEST_ADDRESS),
+          ConfigOverride.config("faban.address", TEST_ADDRESS),
+          ConfigOverride.config("testMode.mockFaban", "true"));
+
+  @Rule
+  public WireMockRule wireMockRule = new WireMockRule(TEST_PORT);
+
+  private MinioService minioServiceSpy;
+  private ExecutorService executorService;
+  private Client client;
+
+
+  @Before
+  public void setUp() throws Exception {
+    minioServiceSpy = Mockito.spy(BenchFlowExperimentManagerApplication.getMinioService());
+    BenchFlowExperimentManagerApplication.setMinioService(minioServiceSpy);
+
+    executorService = BenchFlowExperimentManagerApplication.getExperimentTaskScheduler()
+        .getExperimentTaskExecutorService();
+
+    client = new JerseyClientBuilder(RULE.getEnvironment()).build("test client");
+  }
+
+  @Test
+  public void runScenarioAlwaysCompleted() throws Exception {
+
+    String experimentID = BenchFlowData.SCENARIO_ALWAYS_COMPLETED_EXPERIMENT_ID;
+
+    setUpMocks(experimentID);
+
+    Response response = client.target(String.format("http://localhost:%d", RULE.getLocalPort()))
+        .path(BenchFlowConstants.getPathFromExperimentID(experimentID))
+        .path(BenchFlowExperimentResource.RUN_ACTION_PATH).request().post(null);
+
+    Assert.assertEquals(Response.Status.NO_CONTENT.getStatusCode(), response.getStatus());
+
+    BenchFlowExperimentModelDAO experimentModelDAO = new BenchFlowExperimentModelDAO(mongoClient);
+
+    Assert.assertNotNull(experimentModelDAO.getExperimentModel(experimentID));
+
+    // wait long enough for tasks to start to be executed
+    executorService.awaitTermination(2, TimeUnit.SECONDS);
```

Comment:
I have opened an issue about this since it also applies to the test-manager https://github.com/benchflow/benchflow/issues/395